core/core.gradle.kts

```diff
@@ -66,6 +68,21 @@ tasks.withType<Jar>().configureEach {
     }
 }
 
+tasks.register("generateCudaFile") {
+    description = "Generates a file to let the GRIP runtime know what hardware acceleration to use."
+    doLast {
+        val file = buildDir.resolve("resources/main/META-INF/HardwareAcceleration.mf")
+        file.delete()
+        file.parentFile.mkdirs()
+        file.createNewFile()
+        Files.write("Hardware-Acceleration: ${if (withCuda) "CUDA" else "None"}\n", file, StandardCharsets.UTF_8)
+    }
+}
+
+tasks.withType<JavaCompile>().configureEach {
+    dependsOn("generateCudaFile")
+}
```

Comment:
Might just want to use a properties file in the `resources` directory that you configure using the `WriteProperties` task.
https://docs.gradle.org/current/dsl/org.gradle.api.tasks.WriteProperties.html

Reply:
That looks perfect!